lib/debugger/debugger.c

```diff
@@ -350,6 +361,22 @@ debugger_stop_at_breakpoint(Debugger *self, LogPipe *pipe_, LogMessage *msg)
   return !self->drop_current_message;
 }
 
+gboolean
+debugger_perform_tracing(Debugger *self, LogPipe *pipe_, LogMessage *msg)
+{
+  struct timespec ts, *prev_ts = &self->last_trace_event;
+  gchar buf[1024];
+
+  clock_gettime(CLOCK_MONOTONIC, &ts);
+
+  long diff = prev_ts->tv_sec == 0 ? 0 : timespec_diff_nsec(&ts, prev_ts);
+  printf("[%"G_GINT64_FORMAT".%09"G_GINT64_FORMAT" +%ld] Tracing %s\n",
+         ts.tv_sec, ts.tv_nsec, diff,
+         log_expr_node_format_location(pipe_->expr_node, buf, sizeof(buf)));
```

Comment:
[optional]

I think seeing the source line, like with breakpoints, would be great here as well:
```
  printf("[%"G_GINT64_FORMAT".%09"G_GINT64_FORMAT" +%ld] Tracing %s\n",
         ts.tv_sec, ts.tv_nsec, diff,
         log_expr_node_format_location(pipe_->expr_node, buf, sizeof(buf)));
  _display_source_line(pipe_->expr_node);
```

Reply:
this is just a preliminary stuff, I intend to improve this part a lot later (like random sampling, and backtraces, etc).

for now this is not very useful, because it displays timing information for a single message, this can be affected by jitter a lot, so not very reliable.